src/parser.rs

```diff
@@ -79,32 +79,47 @@ fn parse_return(i: &str) -> IResult<&str, Expr> {
 #[derive(Debug, PartialEq)]
 enum ParsedType {
     BaseType(String),
-    Tuple(Vec<Box<ParsedType>>),
+    Tuple(Box<ParsedType>, Box<ParsedType>),
     Function(Box<ParsedType>, Box<ParsedType>),
 }
 
 fn parse_type_function(i: &str) -> IResult<&str, ParsedType> {
     map(
         tuple((
-            parse_type,
+            alt((parse_type_base, parse_type_tuple)),
```

Comment:
Shouldn't these 2 parsers be the same as `parse_type` below? Otherwise we're indicating that there's some difference between the types of the values we consume vs. the types of the values we produce